### ISSUE TITLE
calculation of track_energy fails on high energy mc "File "/lustre/hpc/hep/pcs557/github/graphnet/src/graphnet/data/extractors/i3truthextractor.py", line 352, in <genexpr>     energy_track = sum(track.total_energy for track in tracks) RuntimeError: "Hadrons" has no mass implemented. Cannot get total energy. (in double I3Particle::GetTotalEnergy() const)"

### ISSUE THREAD

**Issue by RasmusOrsoe** [graphnet-team/graphnet#282]
**Describe the bug**
When the i3TruthExtractor is run on high energy, the calculation fails. The method `_get_primary_track_energy_and_inelasticity` does not generalize outside oscNext MC.

**To Reproduce**
1. Run example i3 conversion script on northern track i3 files (Ask Rasmus for file access)
2. See error

**Expected behavior**
`_get_primary_track_energy_and_inelasticity` should return the correct quantities regardless of energy regime.

**Full traceback**
Please include the full error message to allow for debugging
```
File "make_database.py", line 40, in <module>
    northeren_tracks()
  File "make_database.py", line 35, in northeren_tracks
    converter(paths)
  File "/lustre/hpc/hep/pcs557/github/graphnet/src/graphnet/data/dataconverter.py", line 210, in __call__
    self.execute(filesets)
  File "/lustre/hpc/hep/pcs557/github/graphnet/src/graphnet/data/dataconverter.py", line 277, in execute
    pool = self._iterate_over_individual_files(filesets)
  File "/lustre/hpc/hep/pcs557/github/graphnet/src/graphnet/data/dataconverter.py", line 317, in _iterate_over_individual_files
    self._process_file, tqdm(args, unit="file(s)", colour="green")
  File "/lustre/hpc/hep/pcs557/github/graphnet/src/graphnet/data/dataconverter.py", line 71, in wrapper
    output_file = process_method(self, *args)
  File "/lustre/hpc/hep/pcs557/github/graphnet/src/graphnet/data/dataconverter.py", line 361, in _process_file
    data = self._extract_data(fileset)
  File "/lustre/hpc/hep/pcs557/github/graphnet/src/graphnet/data/dataconverter.py", line 419, in _extract_data
    results = self._extractors(frame)
  File "/lustre/hpc/hep/pcs557/github/graphnet/src/graphnet/data/extractors/i3extractor.py", line 65, in __call__
    return [extractor(frame) for extractor in self]
  File "/lustre/hpc/hep/pcs557/github/graphnet/src/graphnet/data/extractors/i3extractor.py", line 65, in <listcomp>
    return [extractor(frame) for extractor in self]
  File "/lustre/hpc/hep/pcs557/github/graphnet/src/graphnet/data/extractors/i3truthextractor.py", line 152, in __call__
    ) = self._get_primary_track_energy_and_inelasticity(frame)
  File "/lustre/hpc/hep/pcs557/github/graphnet/src/graphnet/data/extractors/i3truthextractor.py", line 352, in _get_primary_track_energy_and_inelasticity
    energy_track = sum(track.total_energy for track in tracks)
  File "/lustre/hpc/hep/pcs557/github/graphnet/src/graphnet/data/extractors/i3truthextractor.py", line 352, in <genexpr>
    energy_track = sum(track.total_energy for track in tracks)
RuntimeError: "Hadrons" has no mass implemented. Cannot get total energy. (in double I3Particle::GetTotalEnergy() const)
```

**Additional context**
It's possible that track energy is already calculated in high energy i3 files. We should talk to domain experts to get a snippet of how track energy is calculated in the high energy regime, or verify that pre-calculated fields are a stable solution.

**Comment by asogaard** [comment]
I think the problem is with the use of the `total_energy` property in `_get_primary_track_energy_and_inelasticity` which assumes that the mass of the particle is set. Another valid implementation (for massless particles) would be to use `energy` instead in the calculation of the inelasticity. I have proposed in the review of https://github.com/graphnet-team/graphnet/pull/283 to add an argument to `_get_primary_track_energy_and_inelasticity` to switch between these two options.

**Comment by asogaard** [comment]
@RasmusOrsoe, was this closed with #283?

**Comment by RasmusOrsoe** [comment]
In a duct-tape fashion, yes.
